subprojects/problems/src/testFixtures/groovy/org/gradle/problems/Closures.groovy

```diff
@@ -30,6 +30,6 @@ class Closures {
         @ClosureParams(FirstParam.class) Closure<?> cl) {
         cl.delegate = target
         cl.resolveStrategy = Closure.DELEGATE_FIRST
-        cl()
+        cl(target)
```

Comment:
This is interesting. I don't understand why the closure should delegate to the target and take the target as the argument. Any idea why we need both?

Reply:
Nope, but consulted with @melix that the fix is sound.

BTW, this is the problem reported is such:

https://ge.gradle.org/s/2enc7pphfskqo/console-log?task=:problems:compileTestFixturesGroovy

Reply:
I understand that the fix works, though another fix would have been to remove the `@ClosureParams(FirstParam.class)`, which seems to be the better option to me. WDYT?

Reply:
`cl()` used to call `cl(null)`, which is calling the closure with a null _parameter_ . By using `cl(target)` we make sure that both the _delegate_ of the closure is set (via `cl.delegate=`) and that if a closure uses an explicit parameter (`{ foo -> ... }`) then the parameter is not null.

Reply:
I do understand that. Though we only use this method in one place, and there the Closure only has `@DelegatesTo`. Moreover, we never use the method for a closure taking a parameter. That is why I'd have opted to remove the parameter altogether.